core/consensus/authority/impl/authority_manager_impl.cpp

```diff
@@ -96,7 +97,65 @@ namespace kagome::authority {
       }
     }
 
-    {  // observe last finalized blocks
+    primitives::AuthorityList authorities;
+    {  // get voter set id at last finalized block
+      const auto &hash = finalized_block_hash;
+      auto header_res = block_tree_->getBlockHeader(hash);
+      if (header_res.has_error()) {
+        log_->critical("Can't get header of block {}: {}",
+                       hash,
+                       header_res.error().message());
+        return false;
+      }
+      const auto &header = header_res.value();
+
+      auto res = trie_storage_provider_->setToEphemeralAt(header.state_root);
```

Comment:
Wait, why do you use trie_storage_provider_, intended for runtime, instead of just TrieStorage?

Reply:
Good point. Thanks